### PR TITLE
fix: access logs buckets objectOwnership to align with S3 Security requirements

### LINKED ISSUE #386: [BUG] (pipeline) accessLogs bucket objectOwnership changes with April 2023 S3 Security Changes
### Describe the bug

There were some S3 Security Changes in April 2023: [Heads-Up: Amazon S3 Security Changes Are Coming in April of 2023](https://aws.amazon.com/blogs/aws/heads-up-amazon-s3-security-changes-are-coming-in-april-of-2023/).

Related CDK issues: [#25288](https://github.com/aws/aws-cdk/issues/25288) and [#25298](https://github.com/aws/aws-cdk/pull/25298)

Docs: [Choosing an Amazon S3 bucket for your standard logs](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AccessLogs.html#AccessLogsBucketAndFileOwnership)

\<quote from docs>
**Important**
Don't choose an Amazon S3 bucket with [S3 Object Ownership](https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html) set to bucket owner enforced. That setting disables ACLs for the bucket and the objects in it, which prevents CloudFront from delivering log files to the bucket.
\</quote from docs>

---

TLDR: for log buckets, `objectOwnership` parameter should be `ObjectOwnership.OBJECT_WRITER`, while when we're creating the accessLogsBucket in PDK [pipeline](https://github.com/aws/aws-prototyping-sdk/blob/mainline/packages/pipeline/src/pdk-pipeline.ts#L109), `ObjectOwnership.BUCKET_OWNER_ENFORCED` is used, which results in the errors mentioned in the above github issues (`InvalidBucketAclWithObjectOwnership`). 

### Expected Behavior

Deploying a `pipeline` project should go through without errors

### Current Behavior

```
PipelineStack | 8:57:46 AM | CREATE_FAILED    | AWS::S3::Bucket      | AccessLogsBucket (AccessLogsBucket83982689) Bucket cannot have ACLs set with ObjectOwnership's BucketOwnerEnforced setting (Service: Amazon S3; Status Code: 400; Error Code: InvalidBucketAclWithObjectOwnership; Request ID: MVCW7FVKN53SMK4N; S3 Extended Request ID: 8ZuYVCf8upqCiU1fR5CJ/jqAYzBwOubsSBLcoRq2pHcYkVPE62E0hlWM6bFi2fleLhhRtOqh25k=; Proxy: null)
```

### Reproduction Steps

Deploy a new `PipelineStack` into a new account

### Possible Solution

`objectOwnership: ObjectOwnership.OBJECT_WRITER,` on `L115`

### Additional Information/Context

_No response_

### PDK version used

0.17.4

### What languages are you seeing this issue on?

Typescript

### Environment details (OS name and version, etc.)

MacOS 13.2.1

### PR DESCRIPTION
**Pull request by sperka**
Fixes #386

### CODE REVIEW

**Review by agdimech — APPROVED** [review]
Thankyou!

**Review by agdimech — APPROVED** [review]
